[@portaljs/components][xl] - @portaljs/ckan package

luccasmmg opened the pull request:
- Renamed `ckan-example` to `ckan-ssg`
- Renamed `learn-example` to `learn`

- Created the `@portaljs/ckan` package, this package provides the following functionality
  - Typescript types for most of the CKAN objects(Datasets, Resources, Organizations, Groups...) 
  - A CKAN Object that has inside a bunch of methods that talk with a predetermined CKAN backend
    - Lets say you want to get all the orgs from a CKAN backend
    - You instancianciate the ckan object with `const ckan = new CKAN("the backend url")`
    - You can then call `const orgs = await ckan.getOrgList()` and the `orgs` variable will already have the correct type
 - Some useful components such as the `<ListOfDatasets>` component that renders a list of all the datasets in a CKAN Backend with built in pagination or the `<DatasetSearchForm>` which renders a form for you to build a search query that can then be used by the `<ListOfDatasets>` component.
 
 - Created an example called `ckan` which has two pages currently
   - An index page that uses the `<DatasetSearchForm>` component + the `<ListOfDatasets>` component to render a simple search page
   - A showcase page that uses the `ckan.getDatasetDetails()` call to get the details of a dataset and them render those in the page

demenech reviewed (changes requested):
Missing dev dependencies
I'm unable to install the project locally and build/publish

demenech commented:
Hey @luccasmmg 
I updated the root folders on the Vercel deployments that were renamed but we are still having some kind error on the deployments

luccasmmg commented:
> Hey @luccasmmg I updated the root folders on the Vercel deployments that were renamed but we are still having some kind error on the deployments

I believe you were seeing an old deployment since the latest one on Vercel is giving me a success message

Below for the learn example
![image](https://github.com/datopian/portaljs/assets/11317382/1d200332-cfa0-442a-bd54-af8f02d35318)

The ckan-ssg example had some problems but i just fixed them
![image](https://github.com/datopian/portaljs/assets/11317382/a86c333b-6a3e-4432-856c-ec559bf20b1c)

demenech commented:
LGTM
